### DISCUSSION TITLE
How to decode AIS targets and write data to log files

### DISCUSSION THREAD

**Discussion by petter5** [antipole2/JavaScript_pi#104] [Ideas]
If using OpenCpn in a patrol boat whee you have a [AIS]receiver and using the VDR , I want to record tracks for all individual vessels /mmsi numbers within the range of the receiver at any time, later be able to find and replay the AIS track message for a single or selected individual vessel(s) / at a given date / time /range and also export these data for this vessel to a file to be used for VDR reply. Can this be done ?

**Comment by antipole2** [comment]
@petter5  Are you seeing the AIS targets in OpenCPN?  If so, then a JavaScript could record them to files - say one file for each MMSI.

What data do you want to record?  It could be anything listed in the plugin User Guide p30 for the OCPNgetAISTargets() API.
But I get the impression you want to record in a format that can be replayed using VDR, so you want NMEA0183 sentences?
What is recorded could easily be changed later.

How up for coding are you?  JavaScript?

I take it you have read @steveads comments on the legal issues?  The solution we might develop here would not be legally water-tight as he indicated.
